zcash_client_sqlite/src/wallet.rs

```diff
@@ -1898,7 +1923,25 @@ pub(crate) fn truncate_to_height<P: consensus::Parameters>(
         }
     }
 
-    // nothing to do if we're deleting back down to the max height
+    // Delete from the scanning queue any range with a start height greater than the
+    // truncation height, and then truncate any remaining range by setting the end
+    // equal to the truncation height + 1. This sets our view of the chain tip back
+    // to the retained height.
+    conn.execute(
+        "DELETE FROM scan_queue
+        WHERE block_range_start >= :new_end_height",
+        named_params![":new_end_height": u32::from(block_height + 1)],
+    )?;
+    conn.execute(
+        "UPDATE scan_queue
+        SET block_range_end = :new_end_height
+        WHERE block_range_end > :new_end_height",
+        named_params![":new_end_height": u32::from(block_height + 1)],
+    )?;
```

Comment:
Why was this moved to earlier in the truncation process?

Reply:
It was previously guarded by the 'if' statement below, meaning that the scan queue was not being correctly updated in all cases. This was one of the bugs fixed by this PR.